src/ORM/ArrayList.php

```diff
@@ -187,29 +186,7 @@ public function toNestedArray()
      */
     public function limit($length, $offset = 0)
     {
-        // Type checking: designed for consistency with DataList::limit()
-        if ((!is_numeric($length) || !is_numeric($offset)) && !is_null($length)) {
-            Deprecation::notice(
-                '4.3',
-                'Arguments to ArrayList::limit() should be numeric or null'
-            );
-        }
-
-        if ($length < 0 || $offset < 0) {
-            Deprecation::notice(
-                '4.3',
-                'Arguments to ArrayList::limit() should be positive'
-            );
-        }
-
         if (!$length) {
-            if ($length === 0) {
-                Deprecation::notice(
```

Comment:
I've raised a new card for implementing this logic - https://github.com/silverstripe/silverstripe-framework/issues/10604